internal/store/postgres/scheduler/job_run_repository.go

```diff
@@ -281,6 +281,113 @@ func (j *JobRunRepository) Create(ctx context.Context, t tenant.Tenant, jobName
 	return errors.WrapIfErr(scheduler.EntityJobRun, "unable to create job run", err)
 }
 
+func (j *JobRunRepository) GetRunSummaryByIdentifiers(ctx context.Context, identifiers []scheduler.JobRunIdentifier) ([]*scheduler.JobRunSummary, error) {
+	if len(identifiers) == 0 {
+		return []*scheduler.JobRunSummary{}, nil
+	}
+
+	var conditions []string
+	var args []any
+	argIndex := 1
+
+	for _, identifier := range identifiers {
```

Comment:
for each run we fetch it from db, this would be very expensive, as in one deep lineage with various scheduled_at we can have dozen of runs. is it better to do it on one query if it's possible

Reply:
or if it's not possible, we can parallelize db call